readthedocs/oauth/models.py

```diff
@@ -134,7 +134,7 @@ class RemoteRepository(models.Model):
         _('SSH URL'),
         max_length=512,
         blank=True,
-        validators=[URLValidator(schemes=['ssh'])],
+        validators=[URLValidator(schemes=['https', 'ssh', 'git'])],
```

Comment:
Why the SSH URL would allow `https` as a scheme on it?

The URL that allows HTTPS is the `clone_url` instead.

Reply:
Opps! I miss read the issue. I'll fix it. Thanks :+1:

Reply:
Did you test that `git@github.com:...` is a valid URL for the URLValidator?

We should, at least, check that URLs from `clone_url` and `ssh_url` (https://developer.github.com/v3/repos/#response) are valid.

We use these at https://github.com/readthedocs/readthedocs.org/blob/041edb30e9fb83e12d15b7238d90cbf9f6b29d8d/readthedocs/oauth/services/github.py#L123-L129.


We should probably check the same for GitLab and Bitbucket.

Reply:
> Opps! I miss read the issue. I'll fix it. Thanks +1

I think the original issue is confusing because it shows a HTTPS URL on the `ssh_url` field, which I thin it's wrong :)

Reply:
> Did you test that `git@github.com:...` is a valid URL for the URLValidator?

Yes :)

GitHub and GitLab uses `git` url for `ssh_url` and Bitbucket uses `ssh` url

GitHub: https://developer.github.com/v3/repos/#get-a-repository
GitLab: https://docs.gitlab.com/ce/api/projects.html
Bitbucket: https://bitbucket.org/api/2.0/repositories/

Reply:
It does not work for me:

```
In [15]: from django.core.validators import URLValidator                                                                                                                                                                                        

In [16]: URLValidator(schemes=['ssh', 'git'])('git@github.com:humitos/readthedocs.org')                                                                                                                                                         
...
ValidationError: ['Enter a valid URL.']

In [17]:
```

Reply:
@humitos I'm sorry I used this scheme to test `git://github.com/user/project-name.git` 
from Django URLValidator docs [(valid URI schemes):](https://docs.djangoproject.com/en/3.0/ref/validators/#django.core.validators.URLValidator.schemes) https://www.iana.org/assignments/uri-schemes/prov/git
Django by default only supports URL's with `://` in the `URLField`

https://github.com/django/django/blob/779e615e362108862f1681f965ee9e4f1d0ae6d2/django/core/validators.py#L105-L107

```python
scheme = value.split('://')[0].lower()
if scheme not in self.schemes:
    raise ValidationError(self.message, code=self.code)
```

Reply:
Right, but that's not the SSH URL that GitHub returns in its API: 

```json
"ssh_url": "git@github.com:octocat/Hello-World.git",
```

> from https://developer.github.com/v3/repos/#list-repositories-for-the-authenticated-user

So, after those object are created, if you go to the admin and try to modify another field, it will fail the validation and won't let you update the other field.

Reply:
Yes, that's right `GitHub` and `GitLab` sends `git@....` URL as `ssh_url`. 
To fix this I think we have to create a Custom URLValidator and override the `__call__` method and copy paste code as we can't call `super()` but not sure if thats a good idea.